src/redux/actions/experiments/updateExperiment.js

```
@@ -47,38 +52,57 @@ const updateExperiment = (
 
       const json = await response.json();
       throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
-
-      dispatch({
-        type: EXPERIMENTS_UPDATED,
-        payload: {
-          experimentId,
-          experiment: experimentDiff,
-        },
-      });
-
-      dispatch({
-        type: EXPERIMENTS_SAVED,
-      });
-    } catch (e) {
-      let { message } = e;
-      if (!isServerError(e)) {
-        console.error(`fetch ${url} error ${message}`);
-        message = endUserMessages.CONNECTION_ERROR;
+    } else if (config.currentApiVersion === api.V2) {
+      if (experimentDiff.sampleIds) {
+        throw new Error('SampleIds in v2 shouldn\'t be updated in this action creator');
       }
```

Can you explain what is your intention behind these lines? Do you mean that this should throw an error incase someone is trying to use the action to update sampleIds?

---

Exactly, updating the `sampleIds` array directly is not possible in any way in the api, this is just a way of making it more clear for developers.

You can either update the samples order through the `/order` endpoint or by hitting the (future) endpoints to add and remove a sample.